### PR TITLE
Operations

### PR DESCRIPTION
**Pull request by victorg1991**

### CODE REVIEW

File: src/http/client.js [hunk 1]
```diff
@@ -30,6 +31,35 @@ export default class HttpClient {
...
+	 * Perform an http request using the given method
+	 * @param {String} method
+	 * @param {String} url
+	 * @param {Object} headers
+	 * @param {Object} body
+	 * @return {Object}
+	 */
+	async request(method, url, headers, body) {
+		const contentType =
+			body instanceof FormData
```
Comment (javiergamarra) [inline src/http/client.js]: Can't it be a property of form (like file) the way of differenciating between form-data. InstanceOf is weird.

File: src/http/client.js [hunk 2]
```diff
@@ -30,6 +31,35 @@ export default class HttpClient {
...
+				? 'multipart/form-data'
+				: 'application/json';
+		const response = await fetch(url, {
+			method,
+			headers: Object.assign(headers, {'Content-Type': contentType}),
```
Comment (javiergamarra) [inline src/http/client.js]: mmm maybe you should use one inmutable library instead of so many object.assign

File: src/http/client.js [hunk 3]
```diff
@@ -42,4 +72,25 @@ export default class HttpClient {
...
+		return JSON.stringify(object);
+	}
+
+	/**
+	 * Buid a form data body with the object provided
+	 * @param {object} object
+	 * @return {object}
+	 */
+	buildFormDataBody(object) {
+		return Object.keys(object).reduce((formData, key) => {
```
Comment (javiergamarra) [inline src/http/client.js:101]: 😂

File: src/http/client.js [hunk 4]
```diff
@@ -60,6 +60,21 @@ export default class HttpClient {
...
+	 * Perform the http request
+	 * @param {string} url
+	 * @param {string} method
+	 * @param {object} headers
+	 * @param {object} body
+	 */
+	doFetch(url, method, headers, body) {
+		return fetch(url, {
+			method,
+			headers: headers,
```
Comment (javiergamarra) [inline src/http/client.js]: :P

**Review by javiergamarra — APPROVED** [review]

File: src/http/client.js (hunk 3 above)
Comment (victorg1991) [reply src/http/client.js:101]: all the things

File: src/http/client.js (hunk 4 above)
Comment (victorg1991) [reply src/http/client.js]: ups

File: src/http/client.js (hunk 1 above)
Comment (victorg1991) [reply src/http/client.js]: how? :(

https://developer.mozilla.org/en-US/docs/Web/API/FormData

File: src/http/client.js (hunk 4 above)
Comment (victorg1991) [reply src/http/client.js]: Done

File: src/http/client.js (hunk 1 above)
Comment (javiergamarra) [reply src/http/client.js]: Yep, it's not feasible, I thought both objects were yours or belonged to the library. One is a string and string checking is horrible, the other is hard to recognize and do a if (body.append) is uglier than the instanceof and more error prone (in that case :P).

Comment (victorg1991) [reply src/http/client.js]: I feel better now: 
https://github.com/wedeploy/wedeploy-sdk-js/blob/master/src/api/node/NodeTransport.js#L126

File: src/http/client.js (hunk 2 above)
Comment (victorg1991) [reply src/http/client.js]: I only use this to copy the contents.

I used to have:

```
const newHeaders = {
...headers,
'Content-Type': contentType,
}
```

Comment (victorg1991) [reply src/http/client.js]: But It needs an extra babel plugin and Liferay's eslint configuration doesn't like it

Comment (javiergamarra) [reply src/http/client.js]: d'oh. I prefer the ... notation. But you are just doing it becase you want a one liner, right?

If you don't want inmutable objects nor one-liners, headers['Content-Type'] = contentType is  clearer.

**Comment by victorg1991** [comment]
I think all the comments are addressed :D

**Comment by javiergamarra** [comment]
👏 👏 👏

**Comment by javiergamarra** [comment]
LGTM, merge :)

**Comment by victorg1991** [comment]
Thanks :D

**Pull request merged**